src/test/ui/span/issue-42234-unknown-receiver-type.rs

```diff
@@ -6,9 +6,8 @@
 // the fix of which this tests).
 
 fn shines_a_beacon_through_the_darkness() {
-    let x: Option<_> = None;
+    let x: Option<_> = None::<_>; //~ ERROR type annotations needed
```

Comment:
Why was the `::<_>` needing to be added here? If that isn't present, does the suggestion point at the binding type?

Reply:
that was back when we only added the `::<X>` suggestion if no explicit generic args were previously present. Getting this test to test the right thing is actually somewhat difficult now :sweat_smile: